src/util/triplea/MapXMLCreator/TerritoryDefinitionsPanel.java

```diff
@@ -0,0 +1,157 @@
+package util.triplea.MapXMLCreator;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.event.MouseEvent;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+class TerritoryDefinitionsPanel extends ImageScrollPanePanel
+{
+	
+	private TerritoryDefinitionsPanel()
+	{
+		s_instance = this;
+	}
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel stepActionPanel)
+	{
+		s_mapXMLCreator = mapXMLCreator;
+		new TerritoryDefinitionsPanel().layout(stepActionPanel);
+	}
+
+	protected void paintCenterSpecifics(final Graphics g, final String centerName, final FontMetrics fontMetrics, final Point item, final int x_text_start)
+	{
+		final HashMap<DEFINITION, Boolean> territoryDefinition = MapXMLHelper.s_territoyDefintions.get(centerName);
+		if (territoryDefinition != null)
+		{
+			final int y_value = item.y + 10;
+			short definition_count = 0;
+			g.setFont(g.getFontMetrics().getFont().deriveFont(Font.BOLD));
+			final FontMetrics fm = g.getFontMetrics();
+			int h = fm.getAscent();
+			for (Entry<DEFINITION, Boolean> definitionEntry : territoryDefinition.entrySet())
+			{
+				if (definitionEntry.getValue())
+				{
+					final int x_value = x_text_start + 17 * definition_count;
+					int w;
+					switch (definitionEntry.getKey())
+					{
+						case IS_WATER:

```

Comment:
Always have a default case in switch statements. Either throw an illegal state exception or log an 'unknown key' warning and print the value of the key. `ClientLogger.java` is a library for logging errors/warnings like that.